rust_src/src/fonts.rs

```diff
@@ -325,23 +336,29 @@ pub fn list_fonts(
     let mut frame: LispFrameRef = frame.into();
 
     let n = match num {
-        Some(n) if n < 0 => return Qnil,
+        Some(n) if n <= 0 => return Qnil,
         Some(n) => n,
         None => 0,
     } as usize;
 
     let list = unsafe { font_list_entities(frame.as_mut(), font_spec.into()) };
+
     match list.into() {
-        Some((car, cdr)) if cdr.is_nil() => match car.as_vector() {
-            Some(vec) if vec.len() == 1 => return list!(vec.get(0)),
-            _ => (),
-        },
-        _ => return Qnil,
+        None => return Qnil,
+        Some((car, cdr)) => {
+            let car_size = match car.as_vector() {
+                Some(v) => v.len(),
+                None => 0,
+            };
+            if (cdr.is_nil() && car_size == 1) {
```

Comment:
If you haven't already, please check the `rustfmt` output. These parentheses are unnecessary:

```
warning: unnecessary parentheses around `if` condition
   --> src/fonts.rs:341:16
    |
341 |             if (cdr.is_nil() && car_size == 1) {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove these parentheses
    |
    = note: #[warn(unused_parens)] on by default
```

Reply:
ack